itest/send_test.go

```diff
@@ -1023,36 +1029,48 @@ func testOfflineReceiverEventuallyReceives(t *harnessTest) {
 // assertAssetSendNtfsEvent asserts that the given asset send event notification
 // was received. This function will block until the event is received or the
 // event stream is closed.
-func assertAssetSendNtfsEvent(t *harnessTest, ctx context.Context,
-	eventNtfns taprpc.TaprootAssets_SubscribeSendAssetEventNtfnsClient,
-	targetEventSelector func(*taprpc.SendAssetEvent) bool,
+func assertAssetSendNtfsEvent(t *harnessTest,
```

Comment:
This func and `assertAssetRecvNtfsEvent` look very similar,  and can be combined with a type constraint + parameter:

```go
type AssetNtfsEvent interface {
	tapdevrpc.SendAssetEvent | tapdevrpc.ReceiveAssetEvent
}

func assertAssetNtfsEvent[T AssetNtfsEvent](t *harnessTest,
	stream *EventSubscription[*T], timeout time.Duration,
	targetEventSelector func(*T) bool, expectedCount int) {
```

From local testing this actually works without specifying the type at the callsites which is kind of neat. Though the timeout arg has to be reordered for these two to match.